saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/DataModel.scala

```diff
@@ -174,7 +178,10 @@ trait DataModel extends Logging {
 
     def apply(f: T => Boolean)(implicit tag: ClassTag[T]): BooleanProperty[T] = {
       def cachedF = if (cache) { x: T => getOrUpdate(x, f).asInstanceOf[Boolean] } else f
-      val a = new BooleanProperty[T](name, cachedF) with NodeProperty[T] { override def node: Node[T] = papply.node }
+      val a = new BooleanProperty[T](name, cachedF) with NodeProperty[T] {
+        override def node: Node[T] = papply.node
+        override val isCachingEnabled: Boolean = isStatic
```

Comment:
Maybe change `isCachingEnabled` to sth closed to `static`? `isStatic` (or `isStaticEnabled`) itself sounds more clear